resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/LoggingPodStatusWatcher.scala

```diff
@@ -92,13 +86,22 @@ private[k8s] class LoggingPodStatusWatcherImpl(
 
   private def closeWatch(): Unit = {
     podCompletedFuture.countDown()
-    scheduler.shutdown()
+    maybeLoggingService.foreach(_.shutdown())
   }
 
-  override def awaitCompletion(): Unit = {
-    podCompletedFuture.await()
-    logInfo(pod.map { p =>
-      s"Container final statuses:\n\n${containersDescription(p)}"
-    }.getOrElse("No containers were found in the driver pod."))
+  override def watchOrStop(sId: String): Unit = maybeLoggingService match {
+    case Some(service) =>
+      logInfo(s"Waiting for application ${conf.appName} with submission ID $sId to finish...")
+      val interval = conf.get(REPORT_INTERVAL)
+      val logShortStatus: Runnable = () => logInfo(s"Application status for $appId (phase: $phase)")
+      service.scheduleAtFixedRate(logShortStatus, 0, interval, TimeUnit.MILLISECONDS)
+      podCompletedFuture.await()
+      logInfo(pod.map { p =>
+        s"Container final statuses:\n\n${containersDescription(p)}"
```

Comment:
since you touched this, the style here is a bit weird. you could do:

```
logInfo(
  pod.map { p => s"Container final statuses:\n\n${containersDescription(p)}" }
    .getOrElse("No containers were found in the driver pod."))
```